mediator/esmFldsExchange_coastal_mod.F90

```diff
@@ -458,8 +557,161 @@ subroutine esmFldsExchange_coastal_init(gcomp, phase, rc)
       end do
       deallocate(S_flds)
     end if
+    
+    
+    !======================================================================
+    ! BEGIN FIELDS TO ICE (compice)
+    !======================================================================
+    ! ---------------------------------------------------------------------                      
+    ! To ice from atm
+    ! ---------------------------------------------------------------------                      
+    ! ----------------------------------------  
+    ! | Atmospheric Fields:                  |
+    ! ----------------------------------------
+    !   - zonal wind height 10m     [m/s]
+    !   - merid wind height 10m     [m/s]
+    !   - lowest atmospheric height [m] (defualt in CICE is 10m see model comp)
+    !   - Temperature at 10m        [k]
+    !   - Specific Humidity         [kg/kg]
+    !   - Pressure at 10m           [Pa]
+    !   - Rainfall rate             [kg/m/m/s]
+    !   - Snowfall rate             [kg/m/m/s]
+    !   - Net shortwave rad         [W/m/m] Other bands are derived from this
+    !   - Net longwave  rad         [W/m/m] 
+    ! ---------------------------------------
+    if (coastal_attr%atm_present .and. coastal_attr%ice_present) then                            
+      allocate(S_flds(14))                                                                       
+      S_flds = (/'Sa_u10m',    & ! inst_zonal_wind_height10m
+                 'Sa_v10m',    & ! inst_merid_wind_height10m                                     
+                 'Sa_z',       & ! lowest atmospheric height                                     
+                 'Sa_t2m',     & ! tenperature at lowest atm level                               
+                 'Sa_q2m',     & ! Specific Humidity                                             
+                 'Sa_pslv',    & ! pressure bottom                                               
+                 'Faxa_rain',  & ! Rain
+                 'Faxa_snow',  & ! Snow
+                 'Faxa_swvdr', & ! Short-wave
+                 'Faxa_swvdf', & ! Short-wave                                                    
+                 'Faxa_swndr', & ! Short-wave                                                    
+                 'Faxa_swndf', & ! Short-wave
+                 'Faxa_swnet', & ! Short-wave net
+                 'Faxa_lwdn' /)  ! long-wave                                                     
+      do n = 1,size(S_flds)
+         fldname = trim(S_flds(n))
+         if (fldchk(is_local%wrap%FBExp(compice),trim(fldname),rc=rc) .and. &
+             fldchk(is_local%wrap%FBImp(compatm,compatm),trim(fldname),rc=rc) &
+            ) then
+            call addmap_from(compatm, trim(fldname), compice, &
+                 mapbilnr_nstod, coastal_attr%mapnorm, coastal_attr%atm2ice_smap) !!mapnstod_consf conservative interp 
```

Comment:
@SmithJos13 these variables are area dependent. Maybe it would be nice to use conservative type interpolation rather than just bilinear. Please check following part from UFS WM - https://github.com/oceanmodeling/CMEPS/blob/9c376237aaecd748f6b552564ed4448d58a58914/mediator/esmFldsExchange_ufs_mod.F90#L77 Of course this also depends on resolution differences between atm, ocn and ice. First order conservative type interpolation could create some patch output and not like smooth bilinear interpolation.